### ISSUE TITLE
Regression: PDO Exceptions when running single tests after #3224

### ISSUE THREAD

**Issue by indigoxela** [backdrop/backdrop-issues#5202]
**Description of the bug**

When trying to run a single test (verified with several local and a tugboat instance), it ends up in a fatal error:

```
ERROR: exception 'PDOException' with message 'SQLSTATE[42S02]: Base table or view not found: 1146 Table 'mydb.simpletest512782semaphore' doesn't exist' in ...core/includes/database/database.inc:2284
```

Different tables, but it always ends up with "Base table or view not found".

Commenting out line 296 in core/includes/config.inc brings back Simpletest functionality. `module_invoke_all('config_create', $config);`

**Steps To Reproduce**

- Enable the locale module
- Try to run a single test - command line or interface in the preview-release or latest dev.

**Actual behavior**

End up with a PDO Exception

**Additional information**

That change is pretty new and not released yet. It's in the 1.20 milestone.

https://github.com/backdrop/backdrop/pull/3695

**Comment by indigoxela** [comment]
Maybe not all tests are affected, but for sure running comment tests is.

Another finding: running tests with `--directory core/modules/whatever` also seems broken.

**Comment by docwilmot** [comment]
docwilmot: I tried running Comment Interface Test on the BackdropCMS.org tugboat demo site, and that also crashed, but the site is running 1.19.3. So maybe this is more than one problem actually.

docwilmot: I didnt get any Exception errors or useful messages, just a test failed and `Undefined property: BackdropWebTestCaseCache::$originalLanguageDefault` in the logs.

docwilmot: This is with and without locale enabled

**Comment by klonos** [comment]
@docwilmot the `Undefined property: BackdropWebTestCaseCache::$originalLanguageDefault` thing should be #4679.

**Comment by docwilmot** [comment]
OK, so definitely two things, but can you confirm, running that test in the demo, now, leads to an AJAX error. Just curious.

**Comment by klonos** [comment]
> ...can you confirm, running that test in the demo, now, leads to an AJAX error.

Confirmed 👍🏼 (separate issue though)

**Comment by indigoxela** [comment]
Bear with me, wild guessing again...

Maybe the execution of hook_config_create() slows down the table-copy-job of Simpletest? So it might be sort of a race condition somewhere?

**Comment by hosef** [comment]
> Maybe the execution of hook_config_create() slows down the table-copy-job of Simpletest?

I don't think that would affect this. The hooks should only be run on the initial install, which happens before the first test is run.

**Comment by indigoxela** [comment]
> The hooks should only be run on the initial install, which happens before the first test is run.

Hm, are you sure? Many setUp() functions install additional modules - core modules or specific testing modules. To my understanding this happens when setting up the individual test, correct me if I'm wrong.

**Comment by hosef** [comment]
That is true, but that doesn't happen until after the tables are all copied.

**Comment by indigoxela** [comment]
Unfortunately this regression breaks [i18n tests](https://github.com/backdrop-contrib/i18n/pull/83/checks?check_run_id=3718606485)

Obviously that modules requires locale to be on. No chance for a workaround other than patching core for the i18n test runs.

Any ideas for a clean solution?

**Comment by indigoxela** [comment]
Wait, no, false alert. It wasn't this problem causing the failures, but just the recent button text changes. All good, sorry for the noise.

**Comment by indigoxela** [comment]
Whatever broke, I'm not able to reproduce it anymore. Closing as outdated.
